doc: Adds example and registry docs for resource_policy (#2623)
mongodb/terraform-provider-mongodbatlas: docs/data-sources/resource_policies.md

# Data Source: mongodbatlas_resource_policies

`mongodbatlas_resource_policies` returns all resource policies in an organization.

> -> **NOTE**: Resource Policies are currently in Public Preview. You will need to

Reviewer: ```suggestion
-> **NOTE**: Resource Policies are currently in Public Preview. To use this feature, you must take the following actions:
```

Author: Changed in a8851acdb8c44933b67c874a3160f5b6938a5ba2

    },
  ]
}

data "mongodbatlas_resource_policy" "project_ip_access_list" {
  org_id = mongodbatlas_resource_policy.project_ip_access_list.org_id
  id     = mongodbatlas_resource_policy.project_ip_access_list.id
}

data "mongodbatlas_resource_policies" "this" {
  org_id = data.mongodbatlas_resource_policy.project_ip_access_list.org_id

  depends_on = [mongodbatlas_resource_policy.project_ip_access_list, mongodbatlas_resource_policy.cloud_provider, mongodbatlas_resource_policy.cloud_region]
}

output "policy_ids" {
  value = { for policy in data.mongodbatlas_resource_policies.this.resource_policies : policy.name => policy.id }
}
```

<!-- schema generated by tfplugindocs -->
## Schema

### Required

- `org_id` (String) Unique 24-hexadecimal digit string that identifies the organization that contains your projects. Use the [/orgs](#tag/Organizations/operation/listOrganizations) endpoint to retrieve all organizations to which the authenticated user has access.

### Read-Only

- `resource_policies` (Attributes List) (see [below for nested schema](#nestedatt--resource_policies))

<a id="nestedatt--resource_policies"></a>
### Nested Schema for `resource_policies`

Read-Only:

> - `created_by_user` (Attributes) The user that last updated the atlas resource policy. (see [below for nested schema](#nestedatt--resource_policies--created_by_user))

Reviewer: ```suggestion
- `created_by_user` (Attributes) The user that last updated the Atlas resource policy. (see [below for nested schema](#nestedatt--resource_policies--created_by_user))
```

Author: Changed in a8851acdb8c44933b67c874a3160f5b6938a5ba2

> - `created_date` (String) Date and time in UTC when the atlas resource policy was created.

Reviewer: ```suggestion
- `created_date` (String) Date and time in UTC when the Atlas resource policy was created.
```

Author: Changed in a8851acdb8c44933b67c874a3160f5b6938a5ba2

> - `id` (String) Unique 24-hexadecimal digit string that identifies an atlas resource policy.

Reviewer: ```suggestion
- `id` (String) Unique 24-hexadecimal digit string that identifies an Atlas resource policy.
```

Author: Changed in a8851acdb8c44933b67c874a3160f5b6938a5ba2

> - `last_updated_by_user` (Attributes) The user that last updated the atlas resource policy. (see [below for nested schema](#nestedatt--resource_policies--last_updated_by_user))

Reviewer: ```suggestion
- `last_updated_by_user` (Attributes) The user that last updated the Atlas resource policy. (see [below for nested schema](#nestedatt--resource_policies--last_updated_by_user))
```

Author: Changed in a8851acdb8c44933b67c874a3160f5b6938a5ba2

> - `last_updated_date` (String) Date and time in UTC when the atlas resource policy was last updated.
> - `name` (String) Human-readable label that describes the atlas resource policy.
> - `org_id` (String) Unique 24-hexadecimal digit string that identifies the organization that contains your projects. Use the [/orgs](#tag/Organizations/operation/listOrganizations) endpoint to retrieve all organizations to which the authenticated user has access.
> - `policies` (Attributes List) List of policies that make up the atlas resource policy. (see [below for nested schema](#nestedatt--resource_policies--policies))
> - `version` (String) A string that identifies the version of the atlas resource policy.

Reviewer: ```suggestion
- `last_updated_date` (String) Date and time in UTC when the Atlas resource policy was last updated.
- `name` (String) Human-readable label that describes the Atlas resource policy.
- `org_id` (String) Unique 24-hexadecimal digit string that identifies the organization that contains your projects. Use the [/orgs](#tag/Organizations/operation/listOrganizations) endpoint to retrieve all organizations to which the authenticated user has access.
- `policies` (Attributes List) List of policies that make up the Atlas resource policy. (see [below for nested schema](#nestedatt--resource_policies--policies))
- `version` (String) A string that identifies the version of the Atlas resource policy.
```

Author: Changed in a8851acdb8c44933b67c874a3160f5b6938a5ba2
